### DISCUSSION TITLE
Is it possible to write an improper list in Gleam?

### DISCUSSION THREAD

**Discussion by CrowdHailer** [gleam-lang/gleam#1018] [Questions & support] [answered]
An improper list is one where the tail of a list is not a list. This [stack overflow answer](https://stackoverflow.com/questions/1919097/functional-programming-what-is-an-improper-list) explains it pretty well.

I can't see how one could be made in Gleam. Yet it looks like the gleam source code supports it.
In the printing code the case of a `NotList` is handled. 
https://github.com/gleam-lang/gleam/blob/main/src/erl.rs#L889

Could anyone share an example of creating some Gleam source that produces a `NotList`?

**Comment by lpil** [comment, marked as answer]
`NotList` here means a variable or something. A more descriptive name might be `NotALiteralList`, but it will still be a list at runtime as the type checker as passed.
```rust
[1, 2, ..this_is_a_variable]
```

All lists have to be proper for the type checker to be happy.

Good use of the questions section! I like this a lot!
